src/viam/sdk/common/service_helper.cpp

```diff
@@ -0,0 +1,42 @@
+#include <viam/sdk/common/service_helper.hpp>
+
+#include <sstream>
+
+namespace viam {
+namespace sdk {
+
+::grpc::Status ServiceHelperBase::fail(::grpc::StatusCode code, const char* message) const noexcept
+    try {
+    std::ostringstream stream;
+    stream << '[' << method_ << "]: " << message;
+    return {code, stream.str()};
+} catch (...) {
```

Comment:
What exception are you `catch`ing here and in methods below? `str()` throwing?

Reply:
More the `<<` operators. But the case must be handled or it isn't noexcept. We could also decide to `abort` in the `catch (...)` case on the argument that we were already in such trouble that we couldn't return a real error.

Reply:
Got it. Current design seems fine to me, then.